### PR TITLE
feat: use named loggers in `coderd`

### PR DESCRIPTION
**Pull request by mtojek**
Related: https://github.com/coder/coder/issues/6726

This PR modifies logger configuration to use named loggers. The idea is to allow DevOps or admins to configure log alarms (catchphrases) for specific issues:

* `[erro]` `coderd` `workspace build error` 
* `[erro]` `coderd.autobuild` `workspace build error` 
* `[erro]` `coderd.provisionerd-%s`
* `[erro]` `coderd.userauth`
* `[erro]` `coderd.prometheusmetrics`

### CODE REVIEW

File: coderd/prometheusmetrics/aggregator.go [hunk 1]
```diff
@@ -22,6 +22,9 @@ const (
 )
 
 const (
+	logPrefix    = "prometheusmetrics: "
+	logPrefixErr = "prometheusmetrics error: "
```
Comment (mafredri) [inline coderd/prometheusmetrics/aggregator.go:26]: Could we use `logger := ma.log.Named("prometheusmetrics"); logger.Debug(...)` instead? I wonder if there's really a need to have the separate err prefix when we already log debug/error levels?

File: coderd/userauth.go [hunk 2]
```diff
@@ -64,6 +68,7 @@ func (api *API) postLogin(rw http.ResponseWriter, r *http.Request) {
 		Email: loginWithPassword.Email,
 	})
 	if err != nil && !xerrors.Is(err, sql.ErrNoRows) {
+		api.Logger.Error(ctx, userauthLogPrefix+"unable to fetch user by email", slog.Error(err))
```
Comment (mafredri) [inline coderd/userauth.go:71]: Same here, perhaps a `logger := api.Logger.Named("...")` would simplify things?

File: coderd/workspaces.go [hunk 3]
```diff
@@ -454,6 +454,10 @@ func (api *API) postWorkspacesByOrganization(rw http.ResponseWriter, r *http.Req
 	}, nil)
 	var bldErr wsbuilder.BuildError
 	if xerrors.As(err, &bldErr) {
+		if bldErr.Status == http.StatusInternalServerError {
```
Comment (mafredri) [inline coderd/workspaces.go:457]: Am I the only one that reads this as "boulder"? 😂

**Review by mafredri — APPROVED** [review]
It's very possible the additional logger errors are failing tests, typically I believe slogtest fails the test if an error level is logged (unless explicitly ignored when creating slogger).

Other than that, looks fine 👍🏻. Suggestions are optional.

**Comment by mtojek** [comment]
> It's very possible the additional logger errors are failing tests, typically I believe slogtest fails the test if an error level is logged (unless explicitly ignored when creating slogger).

Oh, you're right! I totally forgot about this. I assume that we have to disable this feature of slogtest.

File: coderd/prometheusmetrics/aggregator.go (hunk 1 above)
Comment (mtojek) [reply coderd/prometheusmetrics/aggregator.go]: Named loggers can be adopted here. Good idea!

File: coderd/userauth.go (hunk 2 above)
Comment (mtojek) [reply coderd/userauth.go]: Done

**Pull request merged**